intellij/src/test/kotlin/motif/intellij/TestHarness.kt

```diff
@@ -0,0 +1,164 @@
+import com.intellij.testFramework.PsiTestUtil
+import com.intellij.testFramework.fixtures.LightCodeInsightFixtureTestCase
+import motif.Scope
+import motif.core.ResolvedGraph
+import motif.errormessage.ErrorMessage
+import motif.viewmodel.TestRenderer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.File
+import javax.annotation.Nullable
+import javax.inject.Inject
+import kotlin.reflect.KClass
+
+@RunWith(Parameterized::class)
+class TestHarness : LightCodeInsightFixtureTestCase() {
+
+    @get:Rule val rule = IntelliJRule()
+
+    @org.junit.runners.Parameterized.Parameter(0)
+    lateinit var testDir: File
+
+    override fun getProjectDescriptor(): LightProjectDescriptor {
+        return object : ProjectDescriptor(LanguageLevel.HIGHEST) {
+            override fun getSdk(): Sdk? {
+                return JavaAwareProjectJdkTableImpl.getInstanceEx().internalJdk
+            }
+        }
+    }
+
+    @Before
+    public override fun setUp() {
+        super.setUp()
+
+        addLibrary(Inject::class)
+        addLibrary(Scope::class)
+        addLibrary(Nullable::class)
+    }
+
+    @After
+    public override fun tearDown() {
+        super.tearDown()
+    }
+
+    private fun addLibrary(clazz: KClass<*>) {
+        val path = clazz.java.protectionDomain.codeSource.location.path
+        val file = File(path)
+        val libName = file.name
+        PsiTestUtil.addLibrary(myFixture.projectDisposable, myModule, libName, file.parent, libName)
+    }
+
+    @Test
+    fun test() {
+        val testFiles = testDir.walk()
+        val externalFiles = EXTERNAL_ROOT.resolve(testDir.name).walk()
+        (testFiles + externalFiles)
+                .filter { !it.isDirectory && it.name.endsWith(".java") }
+                .forEach { sourceFile ->
+                    myFixture.addClass(sourceFile.readText())
+                }
+        val graph = GraphFactory(project).compute()
+
+        val graphFile = testDir.resolve("GRAPH.txt")
+        if (graphFile.exists()) {
+            val expectedGraphText = graphFile.readText()
+            val actualGraphText = getActualGraphString(graph)
+            assertThat(actualGraphText).isEqualTo(expectedGraphText)
+        }
+
+        val errorFile = testDir.resolve("ERROR.txt")
+        if (errorFile.exists()) {
+            val expectedErrorText = errorFile.readText()
+            val actualErrorText = getActualErrorString(graph)
+            assertThat(actualErrorText).isEqualTo(expectedErrorText)
+        }
+    }
+
+    // TODO Some of this code is duplicated in the compiler TestHarness. Move both test harnesses into the
+    //      same module so they can share code.
+    private fun getActualGraphString(graph: ResolvedGraph): String {
+        val message = TestRenderer.render(graph)
+        val header = "########################################################################\n" +
```

Comment:
nit: multi line strings would be cleaner.

Reply:
Will update